src/components/atoms/Modal/Modal.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, ReactNode, PropsWithChildren } from 'react'
+import * as Dialog from '@radix-ui/react-dialog'
+import classNames from 'classnames'
+import Button, { AppearanceTypes } from 'components/molecules/Button/Button'
+import { ReactComponent as Close } from 'assets/icons/close.svg'
+import { useTranslation } from 'react-i18next'
+
+import classes from './styles.module.scss'
+
+export interface ModalProps {
+  title?: string
+  footer?: ReactNode
+  trigger?: ReactNode
+  customDialogContent?: string
+  topButton?: boolean
+}
+
+const Modal: FC<PropsWithChildren<ModalProps>> = ({
+  title,
+  footer,
+  trigger,
+  children,
+  customDialogContent,
+  topButton,
+}) => {
+  const { t } = useTranslation()
+
+  return (
+    <Dialog.Root>
+      <div>{trigger}</div>
```

Comment:
**[blocker]** (Possibly, maybe not)

Since trigger and footer are React Nodes, then do we need to wrap them in <div> ? Or is this needed for something else ?

Reply:
Yes, we don't need to wrap them, removed